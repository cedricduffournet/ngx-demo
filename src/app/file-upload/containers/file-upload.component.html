<div class="message">
  <input #file type="file" (change)="uploadFile($event)" accept="image/png, image/jpeg">
</div>