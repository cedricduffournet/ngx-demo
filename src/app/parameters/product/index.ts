export * from './parameter-product.component';
