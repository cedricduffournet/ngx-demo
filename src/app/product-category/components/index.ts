export * from './product-category-add/product-category-add.component';
export * from './product-category-update/product-category-update.component';
export * from './product-category-delete/product-category-delete.component';
export * from './product-category-form/product-category-form.component';
export * from './product-category-item/product-category-item.component';
export * from './product-category-items/product-category-items.component';
