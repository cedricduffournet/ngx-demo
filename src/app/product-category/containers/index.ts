export * from './product-category-add-modal/product-category-add-modal.component';
export * from './product-category-update-modal/product-category-update-modal.component';
export * from './product-category-delete-modal/product-category-delete-modal.component';
export * from './product-category-list-view/product-category-list-view.component';
