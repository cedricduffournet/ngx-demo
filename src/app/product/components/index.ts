export * from './product-add/product-add.component';
export * from './product-update/product-update.component';
export * from './product-delete/product-delete.component';
export * from './product-form/product-form.component';
export * from './product-item/product-item.component';
export * from './product-items/product-items.component';
export * from './product-details/product-details.component';
