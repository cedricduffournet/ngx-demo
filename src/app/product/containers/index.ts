export * from './product-add-view/product-add-view.component';
export * from './product-update-modal/product-update-modal.component';
export * from './product-delete-modal/product-delete-modal.component';
export * from './product-list-view/product-list-view.component';
export * from './product-selected/product-selected.component';
export * from './product-details-view/product-details-view.component';
