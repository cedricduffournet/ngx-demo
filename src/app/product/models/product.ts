import { schema } from 'normalizr';
import { ProductCategory, productCategorySchema } from '@app/product-category/models/product-category';

export interface Product {
  id: number;
  name: string;
  categories: ProductCategory[];
}

export const productSchema = new schema.Entity('products', {
  categories: [productCategorySchema]
});
