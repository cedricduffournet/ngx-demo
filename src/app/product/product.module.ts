import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { PipesModule } from '@app/shared/pipes/pipe.module';

import { ProductCategoryModule } from '@app/product-category/product-category.module';
import { ProductEffects } from '@app/product/state/effects';
import { ProductFacade } from '@app/product/state/product.facade';
import {
  ProductAddComponent,
  ProductUpdateComponent,
  ProductDeleteComponent,
  ProductFormComponent,
  ProductItemComponent,
  ProductItemsComponent,
  ProductDetailsComponent
} from '@app/product/components';
import {
  ProductAddViewComponent,
  ProductUpdateModalComponent,
  ProductDeleteModalComponent,
  ProductListViewComponent,
  ProductSelectedComponent,
  ProductDetailsViewComponent
} from '@app/product/containers';
import { reducers } from '@app/product/state/reducers';
import { ProductGuard } from '@app/product/services/product.guard';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    PipesModule,
    ProductCategoryModule,
    EffectsModule.forFeature([ProductEffects]),
    StoreModule.forFeature('products', reducers)
  ],
  declarations: [
    ProductListViewComponent,
    ProductAddComponent,
    ProductUpdateComponent,
    ProductDeleteComponent,
    ProductFormComponent,
    ProductItemComponent,
    ProductItemsComponent,
    ProductAddViewComponent,
    ProductUpdateModalComponent,
    ProductDeleteModalComponent,
    ProductDetailsComponent,
    ProductSelectedComponent,
    ProductDetailsViewComponent
  ],
  entryComponents: [
    ProductUpdateModalComponent,
    ProductDeleteModalComponent
  ],
  exports: [ProductListViewComponent, ProductAddViewComponent, ProductDetailsViewComponent],
  providers: [ProductFacade, ProductGuard]
})
export class ProductModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
