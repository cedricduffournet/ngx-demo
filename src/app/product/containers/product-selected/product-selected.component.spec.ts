import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { provideMockStore } from '@ngrx/store/testing';
import { of } from 'rxjs';
import { BsModalRef } from 'ngx-bootstrap/modal';
import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import { ModalWrapperModule } from '@app/shared/modal';

import { ProductSelectedComponent } from '@app/product/containers';
import { ProductDetailsComponent } from '@app/product/components';
import { ProductFacade } from '@app/product/state/product.facade';
import { AuthFacade } from '@app/authentication/state/auth.facade';
import { PipesModule } from '@app/shared/pipes/pipe.module';
import { ActionsItemsModule } from '@app/shared/actions-item';

describe('ProductSelectedComponent', () => {
  let fixture: ComponentFixture<ProductSelectedComponent>;
  let component: ProductSelectedComponent;
  let facade: ProductFacade;
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        ProductDetailsComponent,
        ProductSelectedComponent
      ],
      imports: [
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule,
        ReactiveFormsModule,
        ModalWrapperModule,
        PipesModule,
        ActionsItemsModule
      ],
      providers: [
        provideMockStore(),
        AuthFacade,
        {
          provide: ProductFacade,
          useValue: {
            selectedDenormalized$: of({
              name: 'Name',
              categories: [
                {
                  id: 1,
                  name: 'Name'
                }
              ]
            }),
            showUpdateProductModal: jest.fn(),
            showDeleteProductModal: jest.fn()
          }
        }
      ]
    });
    facade = TestBed.get(ProductFacade);
    fixture = TestBed.createComponent(ProductSelectedComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should call showUpdateProductModal on update', () => {
    spyOn(facade, 'showUpdateProductModal');
    component.onUpdate();
    expect(facade.showUpdateProductModal).toHaveBeenCalledWith();
  });

  it('should call showDeleteProductModal on delete', () => {
    spyOn(facade, 'showDeleteProductModal');
    component.onDelete();
    expect(facade.showDeleteProductModal).toHaveBeenCalledWith();
  });
});
