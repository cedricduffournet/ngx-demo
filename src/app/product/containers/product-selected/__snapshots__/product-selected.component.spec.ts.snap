// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSelectedComponent should be created 1`] = `
<app-product-selected
  authFacade={[Function AuthFacade]}
  canDelete$={[Function Store]}
  canUpdate$={[Function Store]}
  facade={[Function Object]}
  productFull$={[Function Observable]}
>
  <app-product-details
    ng-reflect-authorization="[object Object]"
    ng-reflect-product="[object Object]"
  >
    <div
      class="product-detail"
    >
      <h1>
        Name
      </h1>
      <h2>
         €
      </h2>
      <p />
      
      <div
        class="category-list"
      >
        <span>
          <strong>
            Categories:
          </strong>
        </span>
        
        <span
          class="category"
        >
           Name  
        </span>
      </div>
      <div
        class="float-right actions"
      >
        <app-actions-item>
          <div
            class="button-validation-action"
          >
            
            
          </div>
        </app-actions-item>
      </div>
    </div>
  </app-product-details>
</app-product-selected>
`;
